### PR TITLE
Add support for NPM two-factor authentication

### LINKED ISSUE #41: npm publish fails when 2fa is enabled for publish
NPM offers an option to enable two factor auth for each publish. When this option is enabled, releasy fails when it tries to publish to npm. The npm CLI returns 401 when it tries to publish and require 2fa, this is perhaps the source of the error, as releasy interprets it as an error and interrupts the execution.

## Your Environment

* Version used: 1.9.1
* Environment name and version (e.g. Chrome 39, node.js 5.4): npm 6.1.0, node 10.1.0
* Operating System and version (desktop or mobile): macOS 10.12.6

### PR DESCRIPTION
**Pull request by lucasecdb**
#### What is the purpose of this pull request?

Adds support for NPM 2FA using a one-time password code, which can now be passed with either the `--otp` flag or by typing it when prompted in the publish step.

#### What problem is this solving?

Resolves #41 

#### How should this be manually tested?

I added tests to cover this functionality, so you can just run `yarn test`.

#### Types of changes
- [x] Bug fix (a non-breaking change which fixes an issue)
- [ ] New feature (a non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Requires change to documentation, which has been updated accordingly.

### CODE REVIEW

File: lib/steps.js [hunk 1]
```diff
@@ -126,22 +127,18 @@ const steps = {
...
     const promise = dryRun
       ? Promise.resolve('')
       : execAsync(cmd).then(({ stdout }) => stdout)
 
-    if (successMessage) {
-      promise.then((stdout) => {
-        if (!quiet) {
-          console.log(chalk`${successMessage} {blue > ${cmd}}`)
-        }
+    const stdout = await promise
```
Comment (estacioneto) [inline lib/steps.js:135]: Why not just await the above expression instead of assigning it to `promise` now? 🤔

File: lib/steps.js [hunk 2]
```diff
@@ -325,19 +322,72 @@ Make your CHANGELOG great again and follow the CHANGELOG format http://keepachan
...
+              }
+
+              steps
+                .publish(config, {
+                  otp: result.otp,
+                  otpMessage: 'Incorrect or expired OTP',
+                  otpRetries: otpRetries - 1,
+                })
+                .then(resolve)
+                .catch(reject)
```
Comment (estacioneto) [inline lib/steps.js:369]: Suuuper nitpick ⛏️ 
```suggestion
                .then(resolve, reject)
```

File: lib/steps.js (hunk 1 above)
Comment (lucasecdb) [reply lib/steps.js:135]: sure! guess I didn't thought of that..

Comment (lucasecdb) [reply lib/steps.js:135]: I thought of doing

```ts
const stdout = await (dryRun
  ? Promise.resolve('')
  : execAsync(cmd).then(({ stdout }) => stdout))
```

but lint complains about the shadowing of `stdout`, what do you think?

File: lib/steps.js (hunk 2 above)
Comment (lucasecdb) [reply lib/steps.js:369]: nice

File: lib/steps.js (hunk 1 above)
Comment (lucasecdb) [reply lib/steps.js:135]: meh, I'll rename the variable inside the `.then`

**Pull request merged**